sdk/eventhub/Azure.Messaging.EventHubs/src/Primitives/EventProcessor{TPartition}.cs

```diff
@@ -0,0 +1,433 @@
+        /// <param name="options">The set of options to use for the processor.</param>
+        ///
+        /// <remarks>
+        ///   If the connection string is copied from the Event Hubs namespace, it will likely not contain the name of the desired Event Hub,
+        ///   which is needed.  In this case, the name can be added manually by adding ";EntityPath=[[ EVENT HUB NAME ]]" to the end of the
+        ///   connection string.  For example, ";EntityPath=telemetry-hub".
+        ///
+        ///   If you have defined a shared access policy directly on the Event Hub itself, then copying the connection string from that
+        ///   Event Hub will result in a connection string that contains the name.
+        /// </remarks>
+        ///
+        protected EventProcessor(int eventBatchSize,
+                                 string consumerGroup,
+                                 string connectionString,
+                                 EventProcessorOptions options = default) : this(eventBatchSize, consumerGroup, connectionString, null, options)
+        {
+        }
+
+        /// <summary>
+        ///   Initializes a new instance of the <see cref="EventProcessor{TPartition}"/> class.
+        /// </summary>
+        ///
+        /// <param name="eventBatchSize">The desired number of events to include in a batch to be processed.  This size is the maximum count in a batch; the actual count may be smaller, depending on whether events are available in the Event Hub.</param>
+        /// <param name="consumerGroup">The name of the consumer group the processor is associated with.  Events are read in the context of this group.</param>
+        /// <param name="connectionString">The connection string to use for connecting to the Event Hubs namespace; it is expected that the shared key properties are contained in this connection string, but not the Event Hub name.</param>
+        /// <param name="eventHubName">The name of the specific Event Hub to associate the processor with.</param>
+        /// <param name="options">The set of options to use for the processor.</param>
+        ///
+        /// <remarks>
+        ///   If the connection string is copied from the Event Hub itself, it will contain the name of the desired Event Hub,
+        ///   and can be used directly without passing the <paramref name="eventHubName" />.  The name of the Event Hub should be
+        ///   passed only once, either as part of the connection string or separately.
+        /// </remarks>
+        ///
+        protected EventProcessor(int eventBatchSize,
+                                 string consumerGroup,
+                                 string connectionString,
+                                 string eventHubName,
+                                 EventProcessorOptions options = default)
+        {
+            Argument.AssertNotNullOrEmpty(consumerGroup, nameof(consumerGroup));
```

Comment:
My general philosophy is that null/empty is a reasonable mistake for initialization but whitespace is usually a bigger issue or intentional.   If someone goes really wants to try to use whitespace, I'm willing to let them try rather than incur the _(tiny)_ hit for the extra validation.